src/Engine/TranskribusClient.php

```diff
@@ -0,0 +1,277 @@
+<?php
+declare(strict_types = 1);
+
+namespace App\Engine;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class TranskribusClient
+{
+
+    /** @var HttpClientInterface */
+    private $httpClient;
+
+    /** @var array transkribus username and password. */
+    private $credentials;
+
+    /** @var int transkribus process ID. */
+    private $processId;
+
+    /** @var string transkribus access token. */
+    private $accessToken = '';
+
+    /** @var string Error message. */
+    private $errorMessage;
+
+    private $hasError = false;
+    
+    /** @var string Transcribed result. */
+    private $textResult = '';
+
+    /** @var string Transcribed result. */
+    private const AUTHENTICATION_URL = 'https://account.readcoop.eu/auth/realms/readcoop/protocol/openid-connect/token';
+
+    /** @var string Transcribed result. */
+    private const PROCESSES_URL = "https://transkribus.eu/processing/v1/processes";
+    
+    /**
+     * TranskribusClient constructor.
+     * @param HttpClientInterface $httpClient
+     * @param array $credentials
+     */
+    public function __construct(
+        HttpClientInterface $httpClient,
+        array $credentials
+    ) {
+        $this->httpClient = $httpClient;
+        $this->credentials = $credentials;
+        $this->setAccessToken();
+    }
+
+     /**
+     * @inheritDoc
+     */
+    private function setAccessToken()
+    {
+        if(file_exists('transkribus_credentials.json')){
+            $data = file_get_contents('transkribus_credentials.json');
+            $content = json_decode($data);
+            $this->accessToken = $content->{'access_token'};
+        }else{
+           $this->retrieveAccessToken();
+        }        
+    }
+
+    private function retrieveAccessToken(){
```

Comment:
The authentication process will be done by #63 so doesn't need to be repeated here.